src/Microsoft.AspNetCore.Rewrite/Internal/IISUrlRewrite/IISUrlRewriteRule.cs

```diff
@@ -57,6 +65,22 @@ public virtual void ApplyRule(RewriteContext context)
                 }
             }
 
+            foreach (ServerVariable serverVariable in ServerVariables)
+            {
+                if (serverVariable.Name.StartsWith(RequestHeaderPrefix, StringComparison.CurrentCultureIgnoreCase))
+                {
+                    context.HttpContext.Request.Headers.Add(
```

Comment:
two of my tests started failing after this change. i decompiled append and see that it no-ops if the key doesn't already exist. going to change the code to add or append as necessary.